eclipse-collections-api/src/main/java/org/eclipse/collections/api/map/MutableMapIterable.java

```diff
@@ -66,6 +67,14 @@ default V add(Pair<? extends K, ? extends V> keyValuePair)
      */
     V removeKey(K key);
 
+    /**
+     * Remove entries from the map at the specified {@code keys}.
+     *
+     * @return true if any entry is removed
+     * @since 10.0
+     */
+    boolean removeAllKeys(Set<? extends K> keys);
+
```

Comment:
More details below, but there should be a default implementation and it should look like this:

```java
    /**
     * Remove entries from the map at the specified {@code keys}.
     *
     * @return @return <tt>true</tt> if this map changed as a result of the call
     * @since 10.0
     */
    default boolean removeAllKeys(Set<? extends K> keys)
    {
        int previousSize = this.size();
        keys.forEach(this::removeKey);
        return previousSize != this.size();
    }
```

I remain skeptical that this method is necessary. Personally, I'd just use keys.forEach(map::removeKey) instead.

Reply:
I want to provide the method on the Map itself. I am fine having the default implementation.